nautobot/extras/jobs.py

```diff
@@ -467,6 +596,12 @@ def validate_data(self, data, files=None):
 
         return f.cleaned_data
 
+    @classmethod
+    def prepare_job_kwargs(cls, job_kwargs):
+        """Process dict and return kwargs that exist as ScriptVariables on this job."""
+        job_vars = cls._get_vars()
+        return {k: v for k, v in job_kwargs.items() if k in job_vars}
```

Comment:
Should we warn or anything if there are things in `job_kwargs` that we're throwing away?

Reply:
I don't think so. The reason we need this is to discard extra fields from forms that the job signature doesn't need. The form should take care of the rest of the data validation.